dataprofiler/data_readers/data_utils.py

```diff
@@ -843,3 +846,138 @@ def url_to_bytes(url_as_string: Url, options: Dict) -> BytesIO:
 
     stream.seek(0)
     return stream
+
+
+class S3Helper:
+    """
+    A utility class for working with Amazon S3.
+
+    This class provides methods to check if a path is an S3 URI
+        and to create an S3 client.
+
+    Attributes:
+        None
+
+    Methods:
+        is_s3_uri(path: str) -> bool: Check if the given path is an S3 URI.
+        create_s3_client(aws_access_key_id: Optional[str] = None,
+                        aws_secret_access_key: Optional[str] = None,
+                        aws_session_token: Optional[str] = None,
+                        region_name: Optional[str] = None) -> boto3.client:
+            Create and return an S3 client.
+        s3_uri_to_bytes(s3_uri: str, s3_client: boto3.client) -> BytesIO:
+            Download an object from an S3 URI and return its content as BytesIO.
```

Comment:
since we have docstrings for all the methods not sure we need this L861-L869

Reply:
same here. deleted the part.